sdk/offchain/src/identifier.rs

```diff
@@ -0,0 +1,386 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::subaddress::{Subaddress, SubaddressParseError};
+use anyhow::{anyhow, Result};
+use bech32::{self, u5, FromBase32, ToBase32};
+use diem_sdk::{
+    move_types::account_address::AccountAddressParseError,
+    types::{
+        account_address::AccountAddress,
+        account_config::allowed_currency_code_string,
+        chain_id::{ChainId, NamedChain},
+    },
+};
+use std::collections::HashMap;
+use thiserror::Error;
+use url::Url;
+
+const DIEM_BECH32_VERSION: u8 = 1;
+
+/// Intent is a struct holdind data decoded from Diem Intent Identifier string
+/// https://dip.diem.com/dip-5/#format
+#[derive(Ord, PartialOrd, Eq, PartialEq, Hash, Clone)]
+pub struct Intent {
+    pub account_address: AccountAddress,
+    pub subaddress: Subaddress,
+    pub currency_code: Option<String>,
+    pub amount: Option<u64>,
+    pub hrp: String,
```

Comment:
is hrp a know acronym? I'd maybe prefer writing out the whole name as it makes reading the code a bit easier.

Reply:
Hrp is human readable part. Hrp is common shortcut used in Bech32 encoding